module/VuFind/src/VuFind/AjaxHandler/GetSideFacets.php

```diff
@@ -227,13 +227,15 @@ protected function formatFacets(
     ) {
         $response = [];
         $facetSet = $recommend->getFacetSet();
+        $facetConfig = $results->getParams()->getFacetConfig();
         foreach ($facets as $facet) {
             if (strpos($facet, ':')) {
                 $response[$facet]['checkboxCount']
                     = $this->getCheckboxFacetCount($facet, $results);
             } else {
                 $context['facet'] = $facet;
                 $context['cluster'] = $facetSet[$facet] ?? [
+                    'label' => $facetConfig[$facet],
```

Comment:
Would it be better to use $results->getParams()->getFacetLabel() here instead of fetching the raw config?

Reply:
I suppose it doesn't hurt, so now changed. I just copied this from \VuFind\Search\Base\Results::buildFacetList. I wonder if it would make sense to use getFacetLabel there as well.

Reply:
Yes, I think it might be a good idea -- but only in the case where the $filter parameter is initially null, since if an override label is provided, we wouldn't want to fall back to the default version. (I can't remember if there's a use case for this, but the function definition seems to imply that it's intentional that you can override default labels via $filter).

Reply:
Maybe the best approach would be to stop overriding $filter when it's null, and instead do something like this at the top of the function:

`$filterFields = array_keys($filter ?? $this->getParams()->getFacetConfig());`

Then the for loop can operate on $filterFields, and when we set the label, we could do it like:

`'label' => $filter[$field] ?? $this->getParams()->getFacetLabel($field)`